lib/semantics/check-coarray.h

```diff
@@ -41,5 +44,40 @@ class CoarrayChecker : public virtual BaseChecker {
       const parser::CharBlock &, parser::MessageFixedText &&);
 };
 
+class CriticalBodyEnforce {
```

Comment:
I'd guess that the class ```CriticalBodyEnforce``` will only be used in check-coarray.cc.  If so, please move it there.

Reply:
Done.